packages/fx-core/resource/yaml.schema.json

```diff
@@ -1133,6 +1134,54 @@
         }
       }
     },
+    "fileUpdateLaunchUrlInLaunchSettings": {
+      "type": "object",
+      "additionalProperties": false,
+      "description": "Update launch url in launchSettings.json. Refer to https://aka.ms/teamsfx-actions/file-updatelaunchurlinlaunchsettings for more details.",
+      "required": ["uses", "with"],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "An optional name of this action."
+        },
+        "env": {
+          "type": "object",
+          "description": "Define environment variables for this action.",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "uses": {
+          "type": "string",
+          "description": "Update launch url in launchSettings.json. Refer to https://aka.ms/teamsfx-actions/file-updatelaunchurlinlaunchsettings for more details.",
+          "const": "file/updateLaunchUrlInLaunchSettings"
+        },
+        "with": {
+          "type": "object",
+          "additionalProperties": false,
+          "description": "parameters for this action",
+          "required": ["target", "profile", "launchUrl", "addLoginHint"],
+          "properties": {
+            "target": {
+              "type": "string",
+              "description": "the target file"
+            },
+            "profile": {
+              "type": "string",
+              "description": "the profile to be updated"
+            },
+            "launchUrl": {
```

Comment:
Instead of the whole url, how about fields like `appId`, `host`, ... to be able to extend for m365 or more features.

Reply:
Done.